src/go/wsl-helper/pkg/wsl-utils/version_windows.go

```diff
@@ -211,87 +210,83 @@ func getPackageVersion(fullName string) (*PackageVersion, error) {
-	)
-	return output, nil
-}
-
 // isInboxWSLInstalled checks if the "in-box" version of WSL is installed,
 // returning whether it's installed, and whether the kernel is installed
-func isInboxWSLInstalled(ctx context.Context) (bool, bool, error) {
-	runWSLExeFunc := runWSLExe
+func isInboxWSLInstalled(ctx context.Context, log *logrus.Entry) (bool, bool, error) {
+	var allErrors []error
+
+	// Check if the core is installed
+	coreInstalled := false
+	newRunnerFunc := NewWSLRunner
 	if f := ctx.Value(&kWSLExeOverride); f != nil {
-		runWSLExeFunc = f.(func(context.Context, ...string) (string, error))
+		newRunnerFunc = f.(func() WSLRunner)
 	}
-
-	output, err := runWSLExeFunc(ctx, "--status")
+	output := &bytes.Buffer{}
+	err := newRunnerFunc().WithStdout(output).Run(ctx, "--status")
 	var exitErr *exec.ExitError
 	if errors.As(err, &exitErr) && exitErr.ExitCode() == wslExitNotInstalled {
-		return false, false, nil
+		// When WSL is not installed, we seem to get exit code 50
 	} else if err != nil {
-		logrus.WithError(err).Trace("wsl.exe --status exited")
-		return false, false, err
-	}
-	lines := strings.Split(strings.TrimSpace(output), "\n")
-	if len(lines) < 1 {
-		return false, false, fmt.Errorf("no output from wsl --status")
+		log.WithError(err).Trace("wsl.exe --status exited")
+		allErrors = append(allErrors, err)
+	} else {
+		lines := strings.Split(strings.TrimSpace(output.String()), "\n")
+		if len(lines) > 0 {
+			coreInstalled = true
+		} else {
+			allErrors = append(allErrors, fmt.Errorf("no output from wsl --status"))
+		}
 	}
 
 	// Check if the kernel is installed.
+	kernelInstalled := false
 	upgradeCodeString := kMsiUpgradeCode
 	if v := ctx.Value(&kUpgradeCodeOverride); v != nil {
 		upgradeCodeString = v.(string)
 	}
 	upgradeCode, err := windows.UTF16PtrFromString(upgradeCodeString)
 	if err != nil {
-		return false, false, err
+		allErrors = append(allErrors, err)
+	} else {
+		productCode := make([]uint16, 39)
+
+		rv, _, _ := msiEnumRelatedProducts.Call(
+			uintptr(unsafe.Pointer(upgradeCode)),
+			uintptr(0),
+			uintptr(0),
+			uintptr(unsafe.Pointer(unsafe.SliceData(productCode))),
+		)
+		switch rv {
+		case uintptr(windows.ERROR_SUCCESS):
+			kernelInstalled = true
+		case uintptr(windows.ERROR_NO_MORE_ITEMS):
+			// kernel is not installed
+		default:
+			err = errorFromWin32("error querying Windows Installer database", rv, nil)
+			allErrors = append(allErrors, err)
+		}
 	}
-	productCode := make([]uint16, 39)
 
-	rv, _, _ := msiEnumRelatedProducts.Call(
-		uintptr(unsafe.Pointer(upgradeCode)),
-		uintptr(0),
-		uintptr(0),
-		uintptr(unsafe.Pointer(unsafe.SliceData(productCode))),
-	)
-	switch rv {
-	case uintptr(windows.ERROR_SUCCESS):
-		return true, true, nil
-	case uintptr(windows.ERROR_NO_MORE_ITEMS):
-		return true, false, nil
-	default:
-		return false, false, errorFromWin32("error querying Windows Installer database", rv, nil)
+	err = errors.Join(allErrors...)
+	if err != nil {
```

Comment:
Again you can wrap the last four lines to `return..., err` but I assume you've chosen not to do this

Reply:
That was just because I've been bitten by https://stackoverflow.com/questions/53892508/ before — but in this case, `err` is always type `error` so it's safe to return it directly.